File: 프로그래머스/lv0/120863. 다항식 더하기/다항식 더하기.py
def solution(polynomial):
    polynomial = polynomial.split(' + ')
    print(polynomial)
    
    a = 0
    b = 0
    
    for i in polynomial:
        if 'x' in i:
            if len(i) > 1:
                a += int(i[:-1])
            else:
                a +=1
        else:
            b += int(i)
    
    if a == 0:
        return f'{b}'
    elif a == 1:
        if b == 0:
            return 'x'
        elif b != 0:
            return f'x + {b}'
    elif a > 1:
        if b == 0:
            return f'{a}x'
        elif b != 0:
            return f'{a}x + {b}'

File: 프로그래머스/0/120863. 다항식 더하기/다항식 더하기.java
class Solution {
    public String solution(String polynomial) {
        String[] terms = polynomial.split(" \\+ ");
        int xCoefficient = 0;
        int constant = 0;

        for (String term : terms) {
            if (term.contains("x")) {
                if (term.equals("x")) {
                    xCoefficient += 1;
                } else {
                    xCoefficient += Integer.parseInt(term.substring(0, term.length() - 1));
                }
            } else {
                constant += Integer.parseInt(term);
            }
        }

        StringBuilder result = new StringBuilder();
        if (xCoefficient == 1) {
            result.append("x");
        } else if (xCoefficient > 1) {
            result.append(xCoefficient).append("x");
        }

        if (constant > 0) {
            if (result.length() > 0) {
                result.append(" + ");
            }
            result.append(constant);
        }

        return result.toString();
    }
}